code/modules/food_and_drinks/machinery/dinner_button.dm

```diff
@@ -0,0 +1,25 @@
+/obj/machinery/foodbutton
+	name = "dinner button"
+	desc = "Casts an announcement to the station, reminding the crew about food."
+	icon = 'icons/obj/service/kitchen.dmi'
+	icon_state = "button_supper"
+	layer = OBJ_LAYER
+	anchored = TRUE
+	pass_flags = PASSTABLE // Able to place on tables
+	var/button_used = FALSE // only one announcement per shift
+	var/button_sound = 'sound/machines/pda_button2.ogg'
+
+/obj/machinery/foodbutton/Initialize(mapload)
+	. = ..()
+
+/obj/machinery/foodbutton/attack_hand(mob/living/user)
+	. = ..()
+	announce()
+
+/obj/machinery/foodbutton/proc/announce(mob/living/user)
+	if(button_used)
+		return FALSE
```

Comment:
```suggestion
		return
	if(world.time - SSticker.round_start_time < STOP_SERVING_BREAKFAST)
		balloon_alert(user, "too early!")
		return
```
We shouldn't let chefs use the button roundstart; it's still breakfast time and they probably haven't cooked much yet.